openshift/deploy/deploymentconfig/postgres-master.yml

```diff
@@ -18,40 +18,56 @@ parameters:
     displayName: Registry
     name: OC_REGISTRY
     required: true
+  - description: Requested CPU
+    displayName: CPU request
+    name: MASTER_CPU_REQUEST
+    required: true
+  - description: Maximum CPU
+    displayName: CPU limit
+    name: MASTER_CPU_LIMIT
+    required: true
+  - description: Requested memory
+    displayName: Memory request
+    name: MASTER_MEMORY_REQUEST
+    required: true
+  - description: Maximum memory
+    displayName: Memory limit
+    name: MASTER_MEMORY_LIMIT
+    required: true
 
 objects:
   - apiVersion: v1
     kind: DeploymentConfig
     metadata:
-      name: ${PREFIX}postgres
+      name: ${PREFIX}postgres-master
     spec:
+      strategy:
+        type: Recreate
       replicas: 1
       selector:
-        name: ${PREFIX}postgres
+        name: ${PREFIX}postgres-master
+      strategy:
+        type: Recreate
       template:
         metadata:
           labels:
-            name: ${PREFIX}postgres
+            name: ${PREFIX}postgres-master
         spec:
           dnsPolicy: ClusterFirst
           restartPolicy: Always
+          terminationGracePeriodSeconds: 3600
           containers:
             - env:
-                - name: POSTGRESQL_USER
-                  valueFrom:
-                    secretKeyRef:
-                      key: database-user
-                      name: ${PREFIX}postgres
-                - name: POSTGRESQL_PASSWORD
+                - name: POSTGRESQL_ADMIN_PASSWORD
                   valueFrom:
                     secretKeyRef:
-                      key: database-password
-                      name: ${PREFIX}postgres
-                - name: POSTGRESQL_DATABASE
+                      key: postgresqlAdminPassword
```

Comment:
there was something that happened where the `-` was not an allowed character...and I should have documented what. bad me.